src/WebJobs.Script/Utility.cs

```diff
@@ -989,17 +989,22 @@ public static bool IsMediaTypeOctetOrMultipart(MediaTypeHeaderValue mediaType)
             }
         }
 
-        // EnableWorkerIndexing set through AzureWebjobsFeatuerFlag always take precdence
-        // if AzureWebjobsFeatuerFlag is not set then WORKER_INDEXING_ENABLED hosting config controls stamplevel enablement
-        // if WORKER_INDEXING_ENABLED is set and WORKER_INDEXING_DISABLED contains the customers app name worker indexing is then disabled for that customer only
-        // Also Worker indexing is disabled for Logic apps
+        // Worker indexing is disabled for Logic apps
+        // EnableWorkerIndexing set through AzureWebjobsFeatureFlag always take precedence
+        // If AzureWebjobsFeatureFlag is not set and
+        // WORKER_INDEXING_DISABLED contains the customers app name worker indexing is then disabled for that customer only
         public static bool CanWorkerIndex(IEnumerable<RpcWorkerConfig> workerConfigs, IEnvironment environment, FunctionsHostingConfigOptions functionsHostingConfigOptions)
         {
             string appName = environment.GetAzureWebsiteUniqueSlotName();
-            bool workerIndexingEnabled = FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagEnableWorkerIndexing, environment)
-                                          || (functionsHostingConfigOptions.WorkerIndexingEnabled
-                                          && !functionsHostingConfigOptions.WorkerIndexingDisabledApps.ToLowerInvariant().Split("|").Contains(appName)
-                                           && !environment.IsLogicApp());
+
+            if (environment.IsLogicApp())
+            {
+                return false;
+            }
+
+            bool workerIndexingFeatureFlagSet = FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagEnableWorkerIndexing, environment);
+            bool workerIndexingDisabledViaHostingConfig = functionsHostingConfigOptions.WorkerIndexingDisabledApps.ToLowerInvariant().Split("|").Contains(appName);
```

Comment:
Can you remind me who sets the Appnames in this hosting config?

Reply:
That would be us. We would be disabling the worker indexing for the function via hosting config.

Reply:
And this hosting config is at stamp level, right?

Reply:
yes